internal/pkg/config/timeouts.go

```diff
@@ -58,4 +59,9 @@ func (c *ServerTimeouts) InitDefaults() {
 
 	// Jitter subtracted from c.CheckinLongPoll.  Disabled if zero.
 	c.CheckinJitter = 30 * time.Second
+
+	// MaxPoll is the maximum allowed value for a long poll when the client specified poll_timeout value is used.
+	// The long poll value is poll_timeout-2m, and the request's write timeout is set to poll_timeout-1m
+	// CheckinMaxPoll values of less then 1m are effectively ignored and a 1m limit is used.
+	c.CheckinMaxPoll = 2 * time.Hour
```

Comment:
@blakerouse @jsoriano added `checkin_max_poll` to timeouts. It only applies to client requested `poll_timeout` values (it does not effect `checkin_long_poll`).
I've extended the config unit test and smoke integration test to verify it works as expected.

Open to feedback for the default value, not sure if we want to use 2h or not

Reply:
Looks good to me, maybe we can start with a lower upper limit. Reducing it later may be breaking.

Should we add this new setting to the documentation?

Reply:
2h seems high, I agree that lowering it would become breaking where raising it would not. So I would start at 1 hour, being its a setting it can be adjusted.